WordPress/openverse-frontend
# Remove usage session cookie

## Fixes


Fixes #791 by @zackkrida

## Description


I ended up choosing not to go with the `localStorage` or `sessionStorage` approaches primarily because those are confined to the _client_ side of the application, meaning that any events happening during the SSR rendering of a page would be impossible to tie to a specific session.

In the linked issue I proposed a solution for being able to tie session IDs together on the analytics server side, essentially by treated each new one generated by the server that is replacing an ID the client already held in session storage as a node in a linked list (the client would be able to create a link between the previous session ID and the new one). That would have the benefit of creating a _longer_ lived session ID than the ephemeral one we have in this PR while still maintaining the basic browser definition of session storage, because if the client doesn't have an ID in session storage then it wouldn't be able to create the link between the previous ID and the new one the server generates.

In any case, I don't know if that level of complexity is necessary yet as we're not even really using the analytics data in any detailed way at the moment as far as I know. The primary goal here, at least in my view, is to be able to continue gathering some analytics data (once we turn the feature back on in production) while still eliminating the cookie that is currently being used.

In the future if we decide we do want that broader full-session ID chain then we can work on an RFC for how to implement that. It will require coordination between the frontend and API to make the changes.

## Testing Instructions

Checkout the branch and run `pnpm dev`. Load the site and ensure analytics requests are sent as expected with a single session ID between the server and client sides of the application.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
